packages/inter-protocol/src/proposals/utils.js

```diff
@@ -68,7 +68,7 @@ export const reserveThenDeposit = async (
   console.info('confirmed deposit for', debugName);
 };
 
-/** @type {<T>(store: ERef<Map<string, T>>, key: string, make: () => T) => Promise<T>} */
+/** @type {<T>(store: ERef<Pick<MapStore<string, T>, 'get' | 'set'>>, key: string, make: () => T) => Promise<T>} */
 const provideWhen = async (store, key, make) => {
   const found = await E(store).get(key);
```

Comment:
I probably should have looked at the actual implementation before.

I was under the impression that `MapStore` threw when doing a `get` of an uninitialized value. That's what I meant earlier by different semantics, unfortunately the signature of `get` methods only varies on the return type between `Map` and `MapStore`, making them "compatible". Thinking more about this, it's probably a footgun that these objects have such a similar shape, yet such different semantics.

Anyway, I think `provideWhen` is faulty because of this, and I don't understand how the fallback path would have every worked with the `MapStore` variant.

Reply:
Agree with your assessment. Is there a requested change in there? An issue to make or a thread to start?

Reply:
I don't know. I don't think it'd be easy enough to fix now, but at least we should create an issue?

I think reverting the `Pick` and adding a `ts-expect-error` pointing to the issue?

Reply:
I made an issue but then realized a simpler fix: 713298

Reply:
I fail to see how it's a fix?